examples/simple.js

```diff
@@ -1,17 +1,238 @@
+        if (inspection.isFulfilled()) {
+          promises_fulfilled = promises_fulfilled +1;
+          if (inspection.value().hasOwnProperty('items')) {
+            prom_with_items = prom_with_items+1; //feeds that had a deal
+            total_items_count = total_items_count+ inspection.value().items.length //total items returned
+            inspection.value().items.forEach(function(item){
+              if (item.hasOwnProperty('upc')) {
+                upc_items_count = upc_items_count+1 //tracking upc_count -- should equal num items in file
+                results_array.push(item)
+              }
+            })
+          }                    
+        }
+      })
+      return results_array;
+    }).then(function(results){
+        //write to file
+        var special_feeds_file = fs.createWriteStream('special_feeds_items.txt');
+        special_feeds_file.on('error', function(err) { console.log(err)/* error handling */ });
+        results.forEach(function(upc_item) { special_feeds_file.write(extractProductInfo(upc_item)); });
+        special_feeds_file.end();
+        
+        //book keeping
+        console.log("Total Feeds Requested: "+category_feed_promises_array.length)
+        console.log("Feeds successfully returned from API: "+promises_fulfilled)
+        console.log("UPC Items Count: "+upc_items_count)
+        console.log("Feeds with Items: "+prom_with_items)
+        console.log("Total Items: "+total_items_count)
+    }).catch(console.error.bind(console))      
+  });
+}
+
+function getSingleListOfPaginatedItems(feed_and_cat_id){
+  
+  //Writes all the feed deals in string representing a feed and category id to a file. 
+  
+  //prepare to use feed and cat_id by splitting them
+  feed = feed_and_cat_id.split(',')[0]
+  cat_id = feed_and_cat_id.split(',')[1]
+  
+  return new Promise(
+    function (resolve, reject) {
+      walmart.paginateByCategory(parseInt(cat_id), feed, function(response) {
+        var results_array = getRecursiveResponse(response);
+        resolve();
+      });
+    }
+  );
+}
+
+function getPaginatedSpecialFeeds(){
+  /*
+    Goes through all valid feed/category combinations and returns Promises of getSingleListOfPaginatedItems
+  */
+  
+  special_feeds_array = [ "specialbuy", "preorder", "clearance" ];
+  //get categories available and retrieve deal information for each one
+  walmart.taxonomy().then(function(result) {
+    var valid_cat_id_array = [];
+    result.categories.forEach(function(category){
+      special_feeds_array.forEach(function(feed){
+        /*
+          go through each feed and prepare string combination of feed and cat_id
+          to retrieve information for.
+        */
+          valid_cat_id_array.push(feed+","+category.id); 
+      })
+    });
+
+    // map clearance call over valid_cat_id_array
+    var category_special_feeds_array = valid_cat_id_array.map(feed_and_cat_id => () => getSingleListOfPaginatedItems(feed_and_cat_id));
+    debugger;
```

Comment:
remove this

Reply:
I'm improving it at the moment.

Reply:
np then. didn't realize

Reply:
Fixed here: 8abf3b3bd3b41232a5185d022ed6d480a4da23e0